cmd/forklift-controller/main.go

```diff
@@ -113,12 +120,98 @@ func main() {
 		log.Error(err, "unable to add kubevirt export APIs to scheme")
 		os.Exit(1)
 	}
+	if err := promv1.AddToScheme(mgr.GetScheme()); err != nil {
+		log.Error(err, "unable to add Prometheus APIs to scheme")
+		os.Exit(1)
+	}
 	if err := template.Install(mgr.GetScheme()); err != nil {
 		log.Error(err, "proceeding without optional OpenShift template APIs")
 	}
 	if err := instancetype.AddToScheme(mgr.GetScheme()); err != nil {
 		log.Error(err, "proceeding without optional kubevirt instance type APIs")
 	}
+
+	mgr.Add(manager.RunnableFunc(func(ctx context.Context) error {
+		log.Info("waiting for cache to sync")
+		if !mgr.GetCache().WaitForCacheSync(ctx) {
+			log.Error(fmt.Errorf("failed to wait for cache sync"), "cache sync failed")
+			return fmt.Errorf("failed to wait for cache sync")
+		}
+
+		clientset, err := kubernetes.NewForConfig(cfg)
+		if err != nil {
+			log.Error(err, "unable to create Kubernetes client")
+			os.Exit(1)
+		}
+
+		log.Info("Setting up Prometheus recording rules")
+
+		namespace := os.Getenv("POD_NAMESPACE")
+		if namespace == "" {
```

Comment:
I thought about it, but in the end its just metric collections, do we really want to fail the controller over it?
also I went with `openshift-mtv` since its the more interested case for us and really the majority of users, but i'm open to suggestions:)